### ISSUE TITLE
Missing resistor?

### ISSUE THREAD

**Issue by jotego** [ika-musume/YMDAC#1]
If you calculate the DAC output impedance for the different input values, you will find that the value is close to R/3 except for `S[2:0]==3'b111`. If you add another resistor of value R before the the first 1/2R resistor, the output impedance becomes exactly R/3 for all S values, which makes more sense.

Keeping a constant output impedance is important in DAC design to prevent non linearities. So I think there is probably a missing R in your schematic.

**Comment by ika-musume** [comment]
I read your comment and drew the Thevenin circuit. As you say, if there is an R connected to +VDC, the impedance will be R/3 at any tap. But that resistor really doesn't exist!

I double-checked the R2R DAC because if the impedance of the R2R DAC is R/2 instead of R, then the impedance at any tap would be R/3. However, the poly resistor elements in the analog shifter and the R2R DAC have same width and height. Unless Yamaha did something special, the resistances should all be the same. 

<img width="1007" alt="YM3012_resistor" src="https://github.com/ika-musume/YMDAC/assets/57454088/1ebde9a8-ab08-431d-907f-3691f4ce0e8d">

The number of resistors connected to +VDC is five, not six that makes all the tap impedance R/3. There is a box that I have marked with "??", but I am not sure if this is a resistor connected to +VDC.

**Comment by ika-musume** [comment]
I read your comment and drew the Thevenin circuit. If there is an R connected to +VDC, as you say, the impedance would be R/3 at any tap. But that resistor really doesn't exist!

I double-checked the R2R DAC, because if the impedance of the R2R DAC is R/2 instead of R, then the impedance at any tap would be R/3. But all poly resistors in the analog shifter and the R2R DAC have the same width and height. Unless Yamaha did something special, the resistance should all be the same.

<img width="1007" alt="YM3012_resistor" src="https://github.com/ika-musume/YMDAC/assets/57454088/c2f7c8d8-2df0-43a6-8a48-081fefe5bf6a">

There is a box that I have marked with "??", I am not sure if this is a resistor connected to +VDC.

**Comment by jotego** [comment]
Thank you for double checking the die shot. The narrow poly stripe marked as _??_ does not seem to be connected from the left side, is it? If it was, the other side is indeed the right net. But the resistance would not be R, as the stripe is too narrow and too long.

I simulated the circuit and indeed the impedance for S[2:0] codes from 7 to 1 is 0.5R, 0.375R, 0.344R, 0.336R and from that point on 0.333R.  I have tried _fixing_ the circuit in different ways with the components present in the schematic (swapping resistors, changing the switch polarity...) But the only solution that makes a constant output impedance is adding an R-valued resistor here.

![image](https://github.com/ika-musume/YMDAC/assets/1863036/91146907-d6bf-463c-a1ce-86c8bd07aff0)

**Comment by ika-musume** [comment]
Yes. The stripe I marked with ?? doesn't look like a resistor at all - only one node is connected.

I looked at the die shot again, but couldn't find the resistor you're talking about. Honestly, I'm starting to wonder why it's not there either. What were they thinking? As I said, it's possible that the impedance of the R2R DAC is 1/2R, but it doesn't seem to be a correct assumption considering the width and height of the poly resistor. I think its impedance is R, resulting in the output impedance of the tap being very strange.

**Comment by jotego** [comment]
I think I'll have to measure it in the lab. This may have been a fault in their design. I'll come back with lab measurements.
